r-package/grf/R/regression_forest.R

```diff
@@ -5,6 +5,10 @@
 #'
 #' @param X The covariates used in the regression.
 #' @param Y The outcome.
+#' @param sample.weights Weights defining the population on which we want our estimator of m(x) to perform well
```

Comment:
I'm not sure this comment is precise, since we don't incorporate sample weights into splitting?

Reply:
I updated the post-processing functions and added tests for them as advised, made the recommended style change in InstrumentalPredictionStrategy, and changed the function signature docs on sample weights to explain how they are used [now sample weighting in causal forest changes the estimand for the ATE/APE, whereas for an HTE they just change the loss] and why they are used [ inverse probability of complete case weighting in censored data]. 

I think that given that weights are pretty conceptually heavy, it may be best to leave the fact that weights aren't used in splitting as an implementation detail. We're still minimizing an estimate of the sample-weighted R-loss, and I think that could be lost on users if we tell them we don't use sample weights in splitting. I'd want to talk about it in a context in which I can say that, and as it stands the function signature docs don't talk about R-loss. Is there a place for that?  If not, maybe future grf.Rd?

Reply:
That sounds good to me -- I think a deeper explanation of weights would make a nice addition to our [algorithm reference](https://github.com/grf-labs/grf/blob/master/REFERENCE.md), it would be great to do that as part of this PR.

Reply:
Done